twilight-model/src/http/reaction_type.rs

```diff
@@ -0,0 +1,109 @@
+use crate::id::{marker::EmojiMarker, Id};
+use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
+use std::fmt::{Display, Formatter, Result as FmtResult};
+
+/// Specifically for HTTP requests, type of [`Reaction`].
+///
+/// [`Reaction`]: crate::channel::message::Reaction
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum RequestReactionType<'a> {
+    /// Custom [`Emoji`].
+    ///
+    /// [`Emoji`]: crate::guild::Emoji
+    Custom {
+        /// Emoji identifier.
+        id: Id<EmojiMarker>,
```

Comment:
I feel as though we should still have the `name` here, even if Discord doesn't end up using it. Who knows, there may come a day it does. The documentation for creating a reaction on a message is:

> To use custom emoji, you must encode it in the format name:id with the emoji name and emoji id.
- <https://discord.com/developers/docs/resources/channel#create-reaction>

This reads to me like the API prefers the name. We should at least give users the ability to provide it so we match the API, regardless of whether it uses it.